File: server/bcephal-settings-service/src/main/java/com/moriset/bcephal/settings/service/IncrementalNumberBrowserData.java
package com.moriset.bcephal.settings.service;

import com.moriset.bcephal.domain.BrowserData;
import com.moriset.bcephal.domain.parameter.IncrementalNumber;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper=false)
public class IncrementalNumberBrowserData extends BrowserData {

	private long initialValue;

	/**
	 * Value by which sequence will increment itself. Increment value can be
	 * positive or negative.
	 */
	private long incrementValue;

	/**
	 * Minimum value of the sequence.
	 */
	private long minimumValue;

	/**
	 * Maximum value of the sequence.
	 */
	private long maximumValue;

	/**
	 * Current value of the sequence.
	 */
	private long currentValue;

	/**
	 * If TRUE : When sequence reaches its set_limit it starts from beginning. IF
	 * FALSE : An exception will be thrown if sequence exceeds its max_value.
	 */
	private boolean cycle;

	/**
	 * Sequence size.
	 */
	private int size;

	/**
	 * Sequence prefix.
	 */
	private String prefix;

	/**
	 * Sequence suffix.
	 */
	private String suffix;
	
	public IncrementalNumberBrowserData(IncrementalNumber incrementalNumber ){
		super(incrementalNumber.getId(), incrementalNumber.getName(),
				incrementalNumber.getCreationDate(), incrementalNumber.getModificationDate());
		setCycle(incrementalNumber.isCycle());
		setIncrementValue(incrementalNumber.getIncrementValue());
		setInitialValue(incrementalNumber.getInitialValue());
		setMaximumValue(incrementalNumber.getMaximumValue());
		setMinimumValue(incrementalNumber.getMinimumValue());
		setSize(incrementalNumber.getSize());
		setPrefix(incrementalNumber.getPrefix());
		setSuffix(incrementalNumber.getSuffix());
		setCurrentValue(incrementalNumber.getCurrentValue());
		setVisibleInShortcut(incrementalNumber.isVisibleInShortcut());
	}
}

File: client/bcephal-client-model/Settings/IncrementalNumberBrowserData.cs
using Bcephal.Models.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Settings
{
    public class IncrementalNumberBrowserData : BrowserData
    {
        public long InitialValue { get; set; }

        public long IncrementValue { get; set; }

        public long MinimumValue { get; set; }

        public long MaximumValue { get; set; }

        public long CurrentValue { get; set; }

        public bool Cycle { get; set; }

        public int Size { get; set; }

        public string Prefix { get; set; }

        public string Suffix { get; set; }

    }
}
